src/daemons/StorageDaemon.cpp

```diff
@@ -114,7 +125,10 @@ int main(int argc, char *argv[]) {
 
     auto handler = std::make_shared<StorageServiceHandler>(kvstore.get(), std::move(schemaMan));
     gServer = std::make_unique<apache::thrift::ThriftServer>();
-    CHECK(!!gServer) << "Failed to create the thrift server";
+    if (!!gServer) {
```

Comment:
By default, `make_unique` could never return a nullptr, instead it will throw bad_alloc on memory allocation failure.

Reply:
Thanks for your reminding.    Good point.  👍